corehq/apps/users/util.py

```diff
@@ -13,7 +13,12 @@
 from django_prbac.utils import ensure_request_has_privilege
 
 
-WEIRD_USER_IDS = ['commtrack-system', 'demo_user']
+WEIRD_USER_IDS = [
+    'commtrack-system',    # internal HQ/commtrack system forms
+    'demo_user',           # demo mode
+    'demo_user_group_id',  # demo mode with case sharing enabled
+]
```

Comment:
This variable is used in a few admin reports and the user elastic index. I'm pretty sure that adding the `demo_user_group_id` only makes those things more correct, though I didn't explicitly dig into that much closer.